lxd/storage/drivers/driver_ceph_utils.go

```diff
@@ -1265,3 +1265,75 @@ func (d *ceph) resizeVolume(vol Volume, sizeBytes int64, allowShrink bool) error
 
 	return err
 }
+
+// findLastCommonSnapshot finds the last common snapshot from the list of snapshots.
```

Comment:
Reading this, its not clear to me what the `snapshots` and `refreshSnapshots` represent, please could you expand the description to explain what they are?

Also I wonder if we should return the index of one of the slices rather than a copy of the Volume, so that we aren't unnecessarily duplicating data?

Reply:
Shall we call `snapshots` `targetSnapshots` instead given thats the terminology used inside the function?

Reply:
I like the idea with the index. Then I would return -1 in case there is no common snapshot.

Reply:
Jup I'll pick a different name for the functions variable alongside the extend comment for explanation.

Reply:
Fixed.